src/SimpleEventStore/SimpleEventStore.AzureDocumentDb.Tests/AzureDocumentDBEventStoreInitializing.cs

```diff
@@ -32,13 +30,20 @@ public async Task when_initializing_all_expected_resources_are_created()
 
             var database = (await client.ReadDatabaseAsync(databaseUri)).Resource;
             var collection = (await client.ReadDocumentCollectionAsync(UriFactory.CreateDocumentCollectionUri(DatabaseName, collectionName))).Resource;
-            var storedProcedure = (await client.ReadStoredProcedureAsync(UriFactory.CreateStoredProcedureUri(DatabaseName, collectionName, TestConstants.AppendStoredProcedureName))).Resource;
+
+            var storedProcedure = client.CreateStoredProcedureQuery(collection.SelfLink)
+                .Where(r => r.Id.StartsWith("appendToStream-"))
```

Comment:
Should we not assert that the correct version of appendToStream is there based on the hash value that the storage engine uses?

Reply:
This one's a case of a pick your poison - the problem is how the test knows about the correct hash.

* Easiest way would be to make the version public, but that's an implementation detail I'd rather not expose in the API surface area.
* Could use `InternalsVisibleToAttribute` to allow the test to access it, but then including that attribute in a library published to NuGet seems naff.  Could put it behind a conditional compilation symbol, but then you've got to update the pipeline to test with a different configuration than it packages with, which doesn't seem worth the effort for this small little thing.
* A third option is reflection, but that has it's own set of complications.
* Could hard code the version in the test, and force it to be manually updated every time the sproc gets updated.

None of the above seem particularly great options.  Any thoughts or ideas?

Reply:
Agree that the options above all have various trade-offs.  Thinking about this, if we have a clean collection per test run then there should only be one stored procedure per test run anyway.

The CI/CD pipeline now uses the CosmosDB emulator, so we get this for free anyway.  I think what you have above is the best option out of the ones we have.